package com.qa.database;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import org.apache.logging.log4j.Logger;
import org.apache.logging.log4j.LogManager;

public class SQLConnector {

	static Logger LOGGER = LogManager.getLogger();

	// Attributes
	private static String localhost;
	private static String user;
	private static String password;

	// Constructor
	public SQLConnector(String user, String password) {
		SQLConnector.localhost = "jdbc:mysql://localhost:3306/storeDB";
		SQLConnector.user = user;
		SQLConnector.password = password;
	}

	public static Connection connectDB() {
		SQLConnector connector = new SQLConnector("root","Apple.Bob1");
		Connection useDB = null;
		try {
			useDB = DriverManager.getConnection(localhost, user, password);
			System.out.println("Connected successfully");
		} catch (SQLException e) {
			LOGGER.debug("Unable to connect");
			LOGGER.debug(e.getStackTrace());
			e.printStackTrace();
		} finally {
			try {
				if(useDB != null) {
					useDB.close();
					System.out.println("Connection complete");
				}
			} catch (SQLException se) {
				se.printStackTrace();
			}
		}
		return useDB;
	}

}
